src/account_manager.rs

```diff
@@ -287,6 +314,24 @@ impl AccountManager {
         }
     }
 
+    /// Sets the password for the stored accounts.
+    pub async fn set_storage_password<P: AsRef<str>>(&mut self, key: P) -> crate::Result<()> {
+        let mut dk = [0; 64];
+        crypto::kdfs::pbkdf::PBKDF2_HMAC_SHA512(key.as_ref().as_bytes(), b"wallet.rs::storage", 100, &mut dk)?;
+        let key: [u8; 32] = dk[0..32][..].try_into().unwrap();
+        *self.storage_encryption_key.lock().await = Some(key);
+
+        let mut accounts = self.accounts.write().await;
+        for encrypted_account in &self.encypted_accounts {
+            let decrypted = crate::storage::decrypt_account_json(encrypted_account, &key)?;
+            let account = serde_json::from_str::<Account>(&decrypted)?;
+            accounts.insert(account.id().clone(), account.into());
+        }
+        self.encypted_accounts = Vec::new();
```

Comment:
```suggestion
        self.encrypted_accounts = Vec::new();
```

Reply:
Are we using `Vec::new()` here because `self.encrypted_accounts.clear()` doesn't shrink the vector?

Reply:
yeah it could be `clear` - missed it :P